qpsolvers/solvers/daqp_.py

```diff
@@ -0,0 +1,182 @@
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# qpsolvers is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with qpsolvers. If not, see <http://www.gnu.org/licenses/>.
+
+"""Solver interface for `DAQP <https://github.com/darnstrom/daqp>`__.
+
+DAQP is a dual active-set algorithm implemented in C implementation. 
+[Arnstrom22]_. It works best small/medium scale dense problems.
+"""
+
+import warnings
+from typing import Optional, Tuple
+
+import numpy as np
+from numpy import linalg as LA 
+from numpy import hstack, vstack
+import daqp
+
+from ..conversions import linear_from_box_inequalities, split_dual_linear_box
+from ..exceptions import ProblemError
+from ..problem import Problem
+from ..solution import Solution
+from ctypes import c_int 
+
+def daqp_solve_problem(
+    problem: Problem,
+    initvals: Optional[np.ndarray] = None,
+    verbose: bool = False,
+    **kwargs,
+) -> Solution:
+    """Solve a quadratic program using daqp.
+
+    Parameters
+    ----------
+    problem :
+        Quadratic program to solve.
+    initvals :
+        Warm-start guess vector (not used).
+    verbose :
+        Set to `True` to print out extra information.
+    Returns
+    -------
+    :
+        Solution to the QP, if found, otherwise ``None``.
+
+    Notes
+    -----
+    """
+    H, f, G, h, A, b, lb, ub = problem.unpack()
+
+    if initvals is not None and verbose:
+        warnings.warn("warm-start values are ignored by daqp")
+
+    Atot: Optional[np.ndarray] = None
+    bupper: Optional[np.ndarray] = None
+    blower: Optional[np.ndarray] = None
+
+    # General constraints
+    if A is not None and b is not None:
+        if G is not None and h is not None:
+            Atot = vstack([G, A])
+            bupper = hstack([h, b])
+        else:
+            Atot = A 
+            bupper = b
+        meq = A.shape[0]
+    else:  # no equality constraint
+        if G is not None and h is not None:
+            Atot = G 
+            bupper  = h
+        meq = 0
+    mineq = bupper.size-meq
+    blower = np.full(bupper.shape,-1e30)
+    # Box constraints
+    if ub is not None:
+        bupper = hstack([ub,bupper])
+        ms = ub.size
+        if lb is not None:
+            blower = hstack([lb,blower])
+        else:
+            blower = hstack([np.full(ms,-1e30),blower])
+    else: #  No upper
+        if lb is not None:
+            ms = lb.size
+            blower = hstack([lb,blower])
+            bupper= hstack([np.full(ms,1e30),bupper])
+        else:
+            ms = 0
+    sense = np.zeros(bupper.shape, dtype=c_int)
+    sense[ms+mineq:] = 5
+
+    x,obj,exitflag,info = daqp.solve(H,f,Atot,bupper,blower,sense)
```

Comment:
This will not pass the CI linter check (should be `, ` rather than `,`). Can you run `black qpsolvers/` on the PR branch?

Reply:
When I ran black on the entire qpsolvers folder it also edited a lot of other files not related to DAQP, so I just ran it for the files that this PR has edited.

Reply:
No worries, this looks good already. If there are some remaining linting errors I'll fix them when merging.